samcli/lib/utils/hash.py

```diff
@@ -0,0 +1,60 @@
+"""
+Hash calculation utilities for files and directories.
+"""
+import os
+import hashlib
+
+BLOCK_SIZE = 4096
+
+
+def file_checksum(file_name):
+    """
+
```

Comment:
can you explain here it is specifically used with `sam deploy` and might potentially be back-compat issues if we break this?

Reply:
Makes sense, I can add a series of comments there.